src/index.ts

```diff
@@ -202,31 +217,42 @@ export async function importActorProfile(
         try {
           if (fileName.endsWith('.json')) {
             result[fileName] = JSON.parse(content)
+            console.log('Parsed JSON file successfully:', fileName)
           } else if (fileName.endsWith('.yaml') || fileName.endsWith('.yml')) {
             result[fileName] = YAML.parse(content)
           } else if (fileName.endsWith('.csv')) {
             result[fileName] = content
+          } else {
+            console.warn(`Unsupported file type: ${fileName}, skipping...`)
           }
         } catch (error: any) {
-          reject(new Error(`Error processing file ${fileName}: ${error}`))
+          console.error(`Error processing file ${fileName}:`, error.message)
```

Comment:
If there is an error, imho this function should `throw` so the caller can catch it.
(the caller might catch it and log, but it also might not want to log).

It could also be useful to allow an `onError` callback to optionally be passed that gets passed with each error. If that is provided, you can call it, it can decide whether to throw or not.

But I definitely think it is always a little risky to `catch` an error and *only* log it. Would be better to throw or pass to a callback.